### PR TITLE
Add resource requests and limits to CSI DaemonSet

### PR DESCRIPTION
**Pull request by chrismuellner**
csi driver now uses:
```yaml
requests:
  cpu: 200m
  memory: 100Mi
limits:
  cpu: 200m
  memory: 100Mi
```

### CODE REVIEW

File: controllers/csi/daemonset.go [hunk 1]
```diff
@@ -210,8 +200,38 @@ func prepareDriverEnvVars() []v1.EnvVar {
 	}
 }
 
-func prepareDriverLivenessProbe() v1.Probe {
-	return v1.Probe{
+func prepareResources() corev1.ResourceRequirements {
+	return corev1.ResourceRequirements{
+		Requests: corev1.ResourceList{
+			corev1.ResourceCPU:    getQuantity(50, resource.Milli),
```
Comment (0sewa0) [inline controllers/csi/daemonset.go:206]: don't use magic numbers

Comment (0sewa0) [reply controllers/csi/daemonset.go:206]: also write test to check for these

File: controllers/csi/daemonset.go [hunk 2]
```diff
@@ -167,41 +162,36 @@ func prepareDriverContainer(operatorImage string) v1.Container {
...
-		LivenessProbe: &livenessProbe,
-		SecurityContext: &v1.SecurityContext{
-			Privileged: &privileged,
-			RunAsUser:  &userID,
-			SELinuxOptions: &v1.SELinuxOptions{
-				Level: "s0",
-			},
-		},
-		VolumeMounts: volumeMounts,
+		Resources:       prepareResources(),
```
Comment (0sewa0) [inline controllers/csi/daemonset.go:174]: 👍 on the refactor

**Review by DTMad — DISMISSED** [review]

**Comment by DTMad** [comment]
Please make the limits and requests the same value to make the Guaranteed QoS pods - see:
https://kubernetes.io/docs/tasks/configure-pod-container/quality-service-pod/#qos-classes

**Review by DTMad — APPROVED** [review]

**Pull request merged**